nemo/collections/asr/parts/preprocessing/perturb.py

```diff
@@ -1011,40 +1202,37 @@ class CustomPerturbation(perturb.Perturbation):
 class AugmentationDataset(IterableDataset):
     """
         A class that loads tarred audio files and cycles over the files in the dataset.
-
         Accepts a single comma-separated JSON manifest file (in the same style as for the AudioToCharDataset/AudioToBPEDataset),
         as well as the path(s) to the tarball(s) containing the wav files. Each line of the manifest should
         contain the information for one audio file, including at least the transcript and name of the audio
         file within the tarball.
-
         Valid formats for the audio_tar_filepaths argument include:
         (1) a single string that can be brace-expanded, e.g. 'path/to/audio.tar' or 'path/to/audio_{1..100}.tar.gz', or
         (2) a list of file paths that will not be brace-expanded, e.g. ['audio_1.tar', 'audio_2.tar', ...].
-
         Note: For brace expansion in (1), there may be cases where `{x..y}` syntax cannot be used due to shell interference.
         This occurs most commonly inside SLURM scripts. Therefore we provide a few equivalent replacements.
         Supported opening braces - { <=> (, [, < and the special tag _OP_.
         Supported closing braces - } <=> ), ], > and the special tag _CL_.
         For SLURM based tasks, we suggest the use of the special tags for ease of use.
-
         See the WebDataset documentation for more information about accepted data and input formats.
     """
 
-    def __init__(self, manifest_path: str, tar_filepaths: Union[str, List[str]], shuffle_n: int = 128):
-        self._manifest = collections.ASRAudioText(manifest_path, parser=parsers.make_parser([]), index_by_file_id=True)
+    def __init__(
+        self,
+        manifest_path: str,
+        tar_filepaths: Union[str, List[str]],
+        shuffle_n: int = 128,
+        rank: int = 0,
+        world_size: int = 1,
+        shard_strategy: str = "replicate",
+    ):
+        from nemo.collections.asr.data.audio_to_text import expand_audio_filepaths
```

Comment:
Hmm is this necessary? Isn't it already imported by other datasets ?

Reply:
You mean is it necessary to have the import inside the constructor versus at the top of the file? We do need that function import, but I can't recall if it needs to be inside the constructor or not.

Reply:
Confirmed, I put the import here in the constructor to avoid a cyclic import, I will add a comment to note this